### PR TITLE
Fix arm64

### PR DESCRIPTION
**Pull request by dcu**
This fixes all of arm64 broken tests and behaviors.

### CODE REVIEW

File: operatorPointwise_unary_test.go [hunk 1]
```diff
@@ -430,12 +430,12 @@ func TestTanhDiff(t *testing.T) {
...
 
-	correct := 1.0 - (math.Tanh(v) * math.Tanh(v)) // I'm surprised Golang doesn't have a secant function!
-	assert.Equal(correct, x.boundTo.(*dualValue).d.Data())
+	correct := 1.0 - (float64(math.Tanh(v)) * float64(math.Tanh(v))) // I'm surprised Golang doesn't have a secant function!
+	assert.InDeltaf(correct, x.boundTo.(*dualValue).d.Data(), 1e-14, "")
 
 	// Tensor edition
 	xdvd := xT.boundTo.(*dualValue).d.(*tensor.Dense)
-	assert.Equal([]float64{correct, correct}, xdvd.Data())
+	assert.InDeltaSlicef([]float64{correct, correct}, xdvd.Data(), 1e-14, "")
```
Comment (owulveryck) [inline operatorPointwise_unary_test.go:438]: This is good enough for the test.
However, if we don't find a proper fix(that validates the equal asertions), we may eventually face another issue that is related. WDYT?

Comment (dcu) [reply operatorPointwise_unary_test.go:438]: I think a difference of 1e-14 should not significant, but let's see what @chewxy thinks about this

File: solvers_test.go [hunk 2]
```diff
@@ -362,7 +362,7 @@ func TestBarzilaiBorweinSolver(t *testing.T) {
 
 	solver := NewBarzilaiBorweinSolver(WithLearnRate(0.0001))
 
-	maxIterations := 200
+	maxIterations := 250
```
Comment (dcu) [inline solvers_test.go:365]: amd64 takes 198 iterations and arm64 takes 210 to reach the expected min cost

File: operatorPointwise_unary_test.go (hunk 1 above)
Comment (chewxy) [reply operatorPointwise_unary_test.go:438]: Can you add a comment on why this is used (i.e. no guarantees of identical behaviours across architecture) - link to https://github.com/golang/go/issues/44528#issuecomment-783975492

Comment (dcu) [reply operatorPointwise_unary_test.go:441]: done

**Pull request merged**